x-pack/solutions/security/plugins/security_solution/public/siem_migrations/common/hooks/use_stop_siem_migration.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMutation } from '@tanstack/react-query';
+import { i18n } from '@kbn/i18n';
+import { useAppToasts } from '../../../common/hooks/use_app_toasts';
+import type { MigrationType } from '../../../../common/siem_migrations/types';
+import { useKibana } from '../../../common/lib/kibana/kibana_react';
+
+export const STOP_SUCCESS = i18n.translate(
+  'xpack.securitySolution.siemMigrations.service.stopMigrationSuccess',
+  { defaultMessage: 'Migration stopped successfully.' }
+);
+export const STOP_ERROR = i18n.translate(
+  'xpack.securitySolution.siemMigrations.service.stopMigrationError',
+  { defaultMessage: 'Error stopping migration.' }
+);
+
+interface StopArgs {
+  migrationId: string;
+}
+
+interface UseStopOptions {
+  onSuccess?: () => void;
+}
+
+export function useStopSiemMigration<T extends MigrationType>(
+  migrationType: T,
+  { onSuccess }: UseStopOptions = {}
+) {
+  const { siemMigrations } = useKibana().services;
+  const { addSuccess, addError } = useAppToasts();
+  return useMutation({
+    mutationKey: ['siemMigration', migrationType, 'stop'],
+    mutationFn: async ({ migrationId }: StopArgs) => {
+      if (migrationType === 'rule') {
```

Comment:
nit: Would be nice to organize it the same way you did in the `useUpdateSiemMigration` hook.

```
const stopMigration =
    migrationType === 'rule'
      ? siemMigrations?.rules?.api?.stopRuleMigration
      : siemMigrations?.dashboards?.api?.stopDashboardMigration;
```